advanced/Scripts/webpage.sh

```diff
@@ -385,19 +387,17 @@ SetWebUILayout() {
 }
 
 CustomizeAdLists() {
-    list="/etc/pihole/adlists.list"
+    local address
+    address="${args[3]}"
 
     if [[ "${args[2]}" == "enable" ]]; then
-        sed -i "\\@${args[3]}@s/^#http/http/g" "${list}"
+        sqlite3 "${gravityDBfile}" "UPDATE adlists SET enabled = 1 WHERE address = '${address}'"
     elif [[ "${args[2]}" == "disable" ]]; then
-        sed -i "\\@${args[3]}@s/^http/#http/g" "${list}"
+        sqlite3 "${gravityDBfile}" "UPDATE adlists SET enabled = 0 WHERE address = '${address}'"
     elif [[ "${args[2]}" == "add" ]]; then
-        if [[ $(grep -c "^${args[3]}$" "${list}") -eq 0 ]] ; then
-            echo "${args[3]}" >> ${list}
-        fi
+        sqlite3 "${gravityDBfile}" "INSERT OR IGNORE INTO adlists (address) VALUES ('${address}')"
```

Comment:
This should probably be `INSERT OR REPLACE` instead of ignore.

Reply:
If we replace then it gets a new added date. Surely, by ignoring, we keep the original added date?

Reply:
Yes. The user wants to insert an adlist, so if the adlist already exists we should either error out or replace the old adlist. It would be better if we threw an error here or checked earlier, but if that is not in place then the adlist should be replaced.

Reply:
Thinking on this again, unless there is some new data, like a comment, being inserted, then we can ignore the insert as the URL is the same.